### ISSUE TITLE
PGNet训练自己的图片（英文字母+数字），最后训练完的模型只显示了数字，英文没有显示是啥问题

### ISSUE THREAD

**Issue by lybcodes** [PaddlePaddle/PaddleOCR#2803]

**Comment by littletomatodonkey** [comment]
看下训练集的情况呢？看下训练集是否也丢字了

**Comment by lybcodes** [comment]
> 看下训练集的情况呢？看下训练集是否也丢字了

训练集也是一样
![捕获](https://user-images.githubusercontent.com/50281525/118426927-bc2dd200-b6fe-11eb-98ad-e815293ec592.PNG)

**Comment by LDOUBLEV** [comment]
预测的文字打印看下有没有英文字母，检查下你的字典都有哪些字符

**Comment by lybcodes** [comment]
> 预测的文字打印看下有没有英文字母，检查下你的字典都有哪些字符

打印出来的也是只有数字，我的字典里是10个数字加26个大写英文字母

**Comment by LDOUBLEV** [comment]
你的代码里修改的地方可以发一下。

检查下字符识别分支的通道数，：https://github.com/PaddlePaddle/PaddleOCR/blob/e59177f07f2ab686e51bcd3f9cf7fe34b30b5e6e/ppocr/modeling/heads/e2e_pg_head.py#L181
检查下CTC_loss输入的lable和f_tcl_char_ld是否是对的：https://github.com/PaddlePaddle/PaddleOCR/blob/e59177f07f2ab686e51bcd3f9cf7fe34b30b5e6e/ppocr/losses/e2e_pg_loss.py#L87

**Comment by lybcodes** [comment]
> 你的代码里修改的地方可以发一下。
> 
> 检查下字符识别分支的通道数，：
> 
> https://github.com/PaddlePaddle/PaddleOCR/blob/e59177f07f2ab686e51bcd3f9cf7fe34b30b5e6e/ppocr/modeling/heads/e2e_pg_head.py#L181
> 
> 
> 检查下CTC_loss输入的lable和f_tcl_char_ld是否是对的：
> https://github.com/PaddlePaddle/PaddleOCR/blob/e59177f07f2ab686e51bcd3f9cf7fe34b30b5e6e/ppocr/losses/e2e_pg_loss.py#L87

![捕获1](https://user-images.githubusercontent.com/50281525/118468425-5578da80-b737-11eb-96ed-dbc1d80a19d4.PNG)
![捕获2](https://user-images.githubusercontent.com/50281525/118468447-5ad62500-b737-11eb-9eeb-3fce062c534e.PNG)

这两个地方我都没动过，我只修改了配置文件跟路径相关的一些参数，如下
Global:
  use_gpu: True
  epoch_num: 400
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/e2e/
  save_epoch_step: 10
  # evaluation is run every 0 iterationss after the 1000th iteration
  eval_batch_step: [ 0, 1000 ]
  # 1. If pretrained_model is saved in static mode, such as classification pretrained model
  #    from static branch, load_static_weights must be set as True.
  # 2. If you want to finetune the pretrained models we provide in the docs,
  #    you should set load_static_weights as False.
  load_static_weights: False
  cal_metric_during_train: False
  pretrained_model: ./en_server_pgnetA/best_accuracy
  checkpoints:
  save_inference_dir:
  use_visualdl: False
  infer_img:
  valid_set: totaltext # two mode: totaltext valid curved words, partvgg valid non-curved words
  save_res_path: ./output/pgnet_r50_vd_totaltext/predicts_pgnet.txt
  character_dict_path: ppocr/utils/number_dict.txt
  character_type: ch
  max_text_length: 50 # the max length in seq
  max_text_nums: 30 # the max seq nums in a pic
  tcl_len: 64

Architecture:
  model_type: e2e
  algorithm: PGNet
  Transform:
  Backbone:
    name: ResNet
    layers: 50
  Neck:
    name: PGFPN
  Head:
    name: PGHead

Loss:
  name: PGLoss
  tcl_bs: 64
  max_text_length: 50 # the same as Global: max_text_length
  max_text_nums: 30 # the same as Global：max_text_nums
  pad_num: 36 # the length of dict for pad

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    learning_rate: 0.001
  regularizer:
    name: 'L2'
    factor: 0


PostProcess:
  name: PGPostProcess
  score_thresh: 0.5
  mode: fast   # fast or slow two ways

Metric:
  name: E2EMetric
  mode: A   # two ways for eval, A: label from txt,  B: label from gt_mat
  gt_mat_dir:  ./train_data/total_text/gt  # the dir of gt_mat
  character_dict_path: ppocr/utils/number_dict.txt
  main_indicator: f_score_e2e

Train:
  dataset:
    name: PGDataSet
    data_dir: ./det6/train
    label_file_list: [./det6/train/train_Label.txt]
    ratio_list: [1.0]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - E2ELabelEncodeTrain:
      - PGProcessTrain:
          batch_size: 14  # same as loader: batch_size_per_card
          min_crop_size: 24
          min_text_size: 4
          max_text_size: 512
      - KeepKeys:
          keep_keys: [ 'images', 'tcl_maps', 'tcl_label_maps', 'border_maps','direction_maps', 'training_masks', 'label_list', 'pos_list', 'pos_mask' ] # dataloader will return list in this order
  loader:
    shuffle: True
    drop_last: True
    batch_size_per_card: 14
    num_workers: 0

Eval:
  dataset:
    name: PGDataSet
    data_dir: ./det6/test
    label_file_list: [./det6/test/test_Label.txt]
    transforms:
      - DecodeImage: # load image
          img_mode: RGB
          channel_first: False
      - E2ELabelEncodeTest:
      - E2EResizeForTest:
          max_side_len: 768
      - NormalizeImage:
          scale: 1./255.
          mean: [ 0.485, 0.456, 0.406 ]
          std: [ 0.229, 0.224, 0.225 ]
          order: 'hwc'
      - ToCHWImage:
      - KeepKeys:
          keep_keys: [ 'image', 'shape', 'polys', 'texts', 'ignore_tags', 'img_id']
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 1 # must be 1
    num_workers: 2

**Comment by lybcodes** [comment]
> 你的代码里修改的地方可以发一下。
> 
> 检查下字符识别分支的通道数，：
> 
> https://github.com/PaddlePaddle/PaddleOCR/blob/e59177f07f2ab686e51bcd3f9cf7fe34b30b5e6e/ppocr/modeling/heads/e2e_pg_head.py#L181
> 
> 
> 检查下CTC_loss输入的lable和f_tcl_char_ld是否是对的：
> https://github.com/PaddlePaddle/PaddleOCR/blob/e59177f07f2ab686e51bcd3f9cf7fe34b30b5e6e/ppocr/losses/e2e_pg_loss.py#L87

请问可以帮看下这个丢字大概是哪里出问题了吗？

**Comment by lybcodes** [comment]
> 你的代码里修改的地方可以发一下。
> 
> 检查下字符识别分支的通道数，：
> 
> https://github.com/PaddlePaddle/PaddleOCR/blob/e59177f07f2ab686e51bcd3f9cf7fe34b30b5e6e/ppocr/modeling/heads/e2e_pg_head.py#L181
> 
> 
> 检查下CTC_loss输入的lable和f_tcl_char_ld是否是对的：
> https://github.com/PaddlePaddle/PaddleOCR/blob/e59177f07f2ab686e51bcd3f9cf7fe34b30b5e6e/ppocr/losses/e2e_pg_loss.py#L87


> 你的代码里修改的地方可以发一下。
> 
> 检查下字符识别分支的通道数，：
> 
> https://github.com/PaddlePaddle/PaddleOCR/blob/e59177f07f2ab686e51bcd3f9cf7fe34b30b5e6e/ppocr/modeling/heads/e2e_pg_head.py#L181
> 
> 
> 检查下CTC_loss输入的lable和f_tcl_char_ld是否是对的：
> https://github.com/PaddlePaddle/PaddleOCR/blob/e59177f07f2ab686e51bcd3f9cf7fe34b30b5e6e/ppocr/losses/e2e_pg_loss.py#L87

我检查了数据集的标签，感觉也没啥问题，不知道这个到底是哪里出错了？？？
![捕获3](https://user-images.githubusercontent.com/50281525/119067600-11c2f100-ba15-11eb-8fb0-426e3d7d634d.PNG)

**Comment by JetHong** [comment]
用官方提供的模型可以识别吗？

**Comment by lybcodes** [comment]
> 用官方提供的模型可以识别吗？

用官方的英文数字模型是可以识别的

**Comment by JetHong** [comment]
> > 用官方提供的模型可以识别吗？
> 
> 用官方的英文数字模型是可以识别的

debug一下，很大可能是生成识别lable的时候没有把英文编码进去（打印一下训练过程中label的字符）

**Comment by lybcodes** [comment]
> > > 用官方提供的模型可以识别吗？
> > 
> > 
> > 用官方的英文数字模型是可以识别的
> 
> debug一下，很大可能是生成识别lable的时候没有把英文编码进去（打印一下训练过程中label的字符）

好的我试下

**Comment by YanfeiZhou1998** [comment]
请问这个问题解决了吗？我也遇到了这个问题。只显示数字没有英文。
